crates/paimon/src/spec/manifest_entry.rs

```diff
@@ -0,0 +1,154 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::spec::DataFileMeta;
+use serde::Deserialize;
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use serde_with::serde_derive::Serialize;
+
+/// Entry representing a file.
+///
+/// Impl Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/manifest/FileEntry.java>
+#[allow(dead_code)]
+pub trait FileEntry {
```

Comment:
I don't think we need a trait for this. Will we have different types of file entries with the same API?

Reply:
Yes, we have a `SimpleFileEntry` to reduce the memory cost in some case.

Reply:
I suggest dividing them into two separate structures and converting them as needed. I believe this approach is much better than storing a `Box<dyn FileEntry>` everywhere.

Reply:
Removed, and keep the `ManifestEntry` now, we can add the `SimpleFileEntry` when used later